docs/src/architecture/data-model.md

```diff
@@ -25,88 +27,103 @@ The figure below illustrats the **data-model** objects and the **inventory**:
+  | **Behavoiur**        | On cloud request<br/>-  provided configuration files are requested from the device and sent to the cloud<br/>- or downloaded from the cloud and sent to the device.<br/><br/> For details see [Configuration Managenement documentation](../references/c8y-configuration-management.md#configuration-files-for-child-devices)
+
+Examples **capability** objects for schema `tedge_config`:
+```json
+"tedge_config": {
+    "files": [
+        { "path": "/etc/tedge/tedge.toml", "type": "tedge.toml" },
+        { "path": "/etc/tedge/mosquitto-conf/c8y-bridge.conf" },
+        { "path": "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf" },
+        { "path": "/etc/mosquitto/mosquitto.conf", "type": "mosquitto", "user": "mosquitto", "group": "mosquitto", "mode": "0o644" }
+    ]
+}
+```
+```json
+"tedge_config": {
+    "files": [
+        { "type": "foo.conf" },
+        { "type": "bar.conf" },
+    ]
+}
+```
 
-* Capability Type: **Logging Management**
+* Capability Schema: **Logging Management**
 
   |                      |              | 
   |:---------------------|:-------------|
   | **Unique name**      | `tedge_log`  |
-  | **Field:** `files`   | List of log-files the device provides |
-  | **Behavoiur**        | One cloud request provided log-files are requested from the device and sent to cloud. Details see TODO: add link to log plugin documentation, section for child-devices.
+  | **Field:** `files`   | TODO |
+  | **Behavoiur**        | TODO |
+
+
+## Inventory API
+
+The inventory is reflected on the MQTT bus under the topic `tedge/inventory`.
+
+* Each device object has it own topic: `/tedge/inventory/<device id>`
+* The `device id` is the `childid` of the **child-device** object, or `main` for the **thin-edge device** object.
+* The payload contains all fields of the device object in JSON format.
+* Example:
+   * topic: `/tedge/inventory/main`
+   * payload:
+```json
+     {
+        "name": "thin-edge device",
+        "type": "thin-edge.io"   
+     }
+```
+* The next level of the topic structure containes the **capability** objects per device:<br/>
+  `/tedge/inventory/<device id>/<capability type>`
+* Example:
+   * topic: `/tedge/inventory/main/tedge_config`
+   * payload:
+```json
+     {
+        "files": [
+           { "path": "/etc/tedge/tedge.toml", "type": "tedge.toml" },
+           { "path": "/etc/tedge/mosquitto-conf/c8y-bridge.conf" },
+           { "path": "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf" },
+           { "path": "/etc/mosquitto/mosquitto.conf", "type": "mosquitto", "user": "mosquitto", "group": "mosquitto", "mode": "0o644" }
+        ]
+     }
+```
+* All messages to `tedge/inventory` and below are published as retain messages.
+  So one who is interested in any object of the inventory can just subscribe to the object's topic and gets directly the object, if it is available in the inventory.
```

Comment:
Although retained messages can somehow help in retrieving all the capabilities, I feel some explicit query APIs (over HTTP or MQTT) would also be needed to improve the UX for an agent developer.